fv3core/stencils/c_sw.py

```diff
@@ -311,174 +302,285 @@ def update_meridional_velocity(
         velocity_c = velocity_c - tmp_flux * flux + rdyc * (ke[0, -1, 0] - ke)
 
 
-@gtstencil
 def initialize_delpc_ptc(delpc: FloatField, ptc: FloatField):
     with computation(PARALLEL), interval(...):
         delpc = 0.0
         ptc = 0.0
 
 
-def vorticitytransport_cgrid(
-    uc: FloatField,
-    vc: FloatField,
-    vort_c: FloatField,
-    ke_c: FloatField,
-    v: FloatField,
-    u: FloatField,
-    dt2: float,
-):
-    """Update the C-Grid zonal and meridional velocity fields.
-
-    Args:
-        uc: x-velocity on C-grid (input, output)
-        vc: y-velocity on C-grid (input, output)
-        vort_c: Vorticity on C-grid (input)
-        ke_c: kinetic energy on C-grid (input)
-        v: y-velocity on D-grid (input)
-        u: x-velocity on D-grid (input)
-        dt2: timestep (input)
+class CGridShallowWaterLagrangianDynamics:
```

Comment:
Are these Lagrangian dynamics? I would suggest removing `Lagrangian` from the name unless you're sure.